Python/Product/Analysis/ProjectEntry.cs

```diff
@@ -36,34 +39,48 @@ namespace Microsoft.PythonTools.Analysis {
     /// </summary>
     [SuppressMessage("Microsoft.Design", "CA1001:TypesThatOwnDisposableFieldsShouldBeDisposable",
         Justification = "Unclear ownership makes it unlikely this object will be disposed correctly")]
-    internal sealed class ProjectEntry : IPythonProjectEntry, IAggregateableProjectEntry {
-        private readonly PythonAnalyzer _projectState;
-        private readonly string _moduleName;
-        private readonly string _filePath;
-        private readonly ModuleInfo _myScope;
-        private IAnalysisCookie _cookie;
-        private PythonAst _tree;
-        private ModuleAnalysis _currentAnalysis;
+    internal sealed class ProjectEntry : IPythonProjectEntry, IAggregateableProjectEntry, IDocument {
         private AnalysisUnit _unit;
-        private int _analysisVersion;
-        private Dictionary<object, object> _properties = new Dictionary<object, object>();
         private ManualResetEventSlim _curWaiter;
         private int _updatesPending, _waiters;
+        private readonly SortedDictionary<int, DocumentBuffer> _buffers;
         internal readonly HashSet<AggregateProjectEntry> _aggregates = new HashSet<AggregateProjectEntry>();
```

Comment:
Can be made private

Reply:
I haven't touched aggregates at all in this PR and I'd rather not start now.